File: python/triangle.py
def is_triangle(sides):
    [a, b, c] = sorted(sides)
    return a > 0 and a + b >= c


def size(sides):
    return len(set(sides))


def equilateral(sides):
    return is_triangle(sides) and size(sides) == 1


def isosceles(sides):
    return is_triangle(sides) and size(sides) != 3


def scalene(sides):
    return is_triangle(sides) and size(sides) == 3

File: clojure/triangle.clj
(ns triangle)

(defn- different-sides [& xs] (count (set xs)))

(defn is-valid? [a b c] (and (> (+ a b) c) (> (+ a c) b) (> (+ b c) a)))

(defn equilateral? [a b c] (and (is-valid? a b c) (= a b c)))

(defn isosceles? [a b c] (and (is-valid? a b c) (>= 2 (different-sides a b c))))

(defn scalene? [a b c] (and (is-valid? a b c) (= 3 (different-sides a b c))))
